### DISCUSSION TITLE
Unsure how to get network.hostname() working in RPi Pico

### DISCUSSION THREAD

**Discussion by mikespolk** [micropython/micropython#15562] [RP2040 / Pico] [answered]
I am having a tiny web server in RPi PicoW. To connect it to WLAN, I have the following commands in the Micropython script:
    import network
    network.hostname("pico")
    wlan.connect(ssid, password)
When started, I can connect to RPi with browser, using address http://192.168.1.12 but when trying to use http://pico, browser connection does not open.

I have double-checked the problem with ping command in Win11 laptop, by trying each of "pico", "pico." and "pico.local" as destination address, but I get error "Ping request could not find host pico."

Weird thing is that IMO the DNS name has succeeded more than one time in the past weeks. However, I cannot get it work anymore. As WLAN hub, I have tested both ZyXEL LTE 3301 and a Samsung phone WLAN hotspot.

How to get the DNS destination name working?

**Comment by scruss** [comment]
You're doing hostname correctly it's just that mDNS doesn't work as it should: [mDNS support on Pico · micropython · Discussion #10932](https://github.com/orgs/micropython/discussions/10932)

**Comment by mikespolk** [comment]
Ah, ok, thank you...
Still I don't understand what might accurately speaking be the problem. Is it dependent of my router type, so that it might not support such feature? Or should I still correct something in my RPi Pico Micropython script?

**Reply by scruss** [reply, marked as answer]
No, there's nothing you can do until the issue's fixed.

Depending on the domain your router uses, you might be able to access it through your router's search domain. In my case, that would be a host like **pico.ht.home**, but yours is probably different

**Comment by mikespolk** [comment]
Thank you for advice, Stewart!
